### PR TITLE
Update rawtherapee to 4.2.1025

### LINKED ISSUE #24960: Rawtherapee.rb should be updated
brew cask install --force rawtherapee
==> Satisfying dependencies
complete
==> Downloading http://rawtherapee.com/releases_head/mac/RawTherapee_OSX_10.11_64_4.2-876_master.zip
######################################################################## 100.0%
curl: (22) The requested URL returned error: 404 Not Found
Error: Download failed on Cask 'rawtherapee' with message: Download failed: http://rawtherapee.com/releases_head/mac/RawTherapee_OSX_10.11_64_4.2-876_master.zip

It seems that rawtherapee was updated to 4.2.1025 and https://github.com/caskroom/homebrew-cask/blob/master/Casks/rawtherapee.rb should be modified:
version '4.2.876' -> version '4.2.1025'
and sha256 should be updated accordingly

### PR DESCRIPTION
**Pull request by frhgit**
#### Checklist
- [x] The commit message includes the cask’s name and version.
- [x] `brew cask audit --download {{cask_file}}` is error-free.
- [x] `brew cask style --fix {{cask_file}}` left no offenses.

Closes https://github.com/caskroom/homebrew-cask/issues/24960.

### CODE REVIEW

**Comment by vitorgalvao** [comment]
Thank you.

**Review by vitorgalvao — APPROVED** [review]

File: Casks/rawtherapee.rb [hunk 1]
```diff
@@ -25,7 +25,7 @@
   elsif MacOS.version <= :yosemite
     container nested: "RawTherapee_OSX_10.10_64_#{version}.dmg"
   else
-    container nested: "RawTherapee_OSX_10.11_64_#{version.major_minor}-#{version.patch}_master.dmg"
+    container nested: "RawTherapee_OSX_10.11_64_#{version}.dmg"
```
Comment (reitermarkus) [inline Casks/rawtherapee.rb]: This should be `"RawTherapee_OSX_10.11_64_#{version}/RawTherapee_OSX_10.11_64__.dmg"`.

**Review by reitermarkus — CHANGES_REQUESTED** [review]

File: Casks/rawtherapee.rb (hunk 1 above)
Comment (reitermarkus) [inline Casks/rawtherapee.rb]: Also, could you move the `container` stanzas to the `if` statement above?

Comment (frhgit) [inline Casks/rawtherapee.rb]: I would to those changes with pleasure, but I have no clue how to that. In short I have never used github :/

Comment (frhgit) [inline Casks/rawtherapee.rb]: Thank you.
Tell me if you are ok with the last change

File: Casks/rawtherapee.rb [hunk 2]
```diff
@@ -24,8 +24,7 @@
     container nested: "RawTherapee_OSX_10.6_64_#{version}.dmg"
   elsif MacOS.version <= :yosemite
     container nested: "RawTherapee_OSX_10.10_64_#{version}.dmg"
-  else
```
Comment (reitermarkus) [inline Casks/rawtherapee.rb]: You accidentally deleted the else clause here. Either add this back or move all of the `container nested` lines in the respective branches of the existing `if` statement above.

**Review by reitermarkus — CHANGES_REQUESTED** [review]

**Review by reitermarkus — APPROVED** [review]

**Pull request merged**